### PR TITLE
Add more guidance on Ruby

### PR DESCRIPTION
**Pull request by gminn**
Plus a GitHub workflow to make sure the `README` and `.ruby-version` are in sync 😎 

Can be tested by running the workflow manually:

```
gh workflow run detect-ruby-version-mismatch.yml --ref $(git rev-parse --abbrev-ref HEAD)
```

Tested by checking for failure when either the `README` or the `.ruby-version` was changed and there was a mismatch. Checked for success by reverting each change. Results are [here](https://github.com/memfault/interrupt/actions/workflows/detect-ruby-version-mismatch.yml).

Sadly didn't know re-named workflows generate a new workflow and [can't get removed](https://github.com/orgs/community/discussions/26256)...so there is an extra workflow in [Actions](https://github.com/memfault/interrupt/actions) from when I renamed the workflow that I've disabled for now.

### CODE REVIEW

**Review by noahp — APPROVED** [review]
nice!

**Review by mrcljx — APPROVED** [review]
Feel free to move ahead.

File: .github/workflows/detect-ruby-version-mismatch.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+      - 'readme.md'
+      - '.ruby-version'
+
+jobs:
+  check_ruby_version:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v3
+      - name: Get version of Ruby
```
[the remark this answers is not in the document]
Comment (gminn) [reply .github/workflows/detect-ruby-version-mismatch.yml]: Good call! Done

File: readme.md [hunk 2]
```diff
@@ -21,15 +21,43 @@ See [Contributing](https://interrupt.memfault.com/contributing) for more informa
 
 ### Locally
 
-Follow the instructions in the [Jekyll quickstart guide](https://jekyllrb.com/docs/) to install Ruby and Jekyll and bundler.
+You'll need:
+- Python 3.8 or later
+- Ruby 2.7.7
```
[the remark this answers is not in the document]
Comment (gminn) [reply readme.md:26]: Fair point, but then I wouldn't get to force a workflow to fail, which is fun 😀

**Pull request merged**